Automattic/jetpack
# Carousel: prevent interaction on transition

#### Changes proposed in this Pull Request

Hello!

If you're like me, and love clicking fast you'd have noticed that clicking at great pace on the next and previous buttons of a gallery carousel sometimes triggers the built-in double click zoom effect.

To prevent that from happening, this PR activates a Swiper JS parameter `preventInteractionOnTransition` **for non-touch devices**. Quoting the API docs:

> When enabled it won't allow to change slides by swiping or navigation/pagination buttons during transition

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/6458278/127265716-f4d7fe14-4381-4785-8d1a-c2e72e880345.gif" width="250" /> | <img src="https://user-images.githubusercontent.com/6458278/127265709-213eb9aa-75fe-4731-a45c-505bf6343824.gif" width="250" /> |

You're still permitting to click fast!


#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions

Fire up a gallery in a desktop browser (a non-touch device).

⏱️ Proceed to click as fast as you can. Next, previous, go for it!

The image shouldn't zoom.

Try on a touch device. You should be able to swipe as expected.

Other events such as pinch and zoom and double tap (on the image) should also work as expected.
